boilerplate/openshift/golang-osd-operator/csv-generate/csv-generate.sh

```diff
@@ -87,13 +87,17 @@ if [[ "${RELEASE_BRANCHED_BUILDS}" ]]; then
     BRANCH="release-${operator_version%.*}"
 fi
 
-git clone ${GIT_PATH} "$SAAS_OPERATOR_DIR"
-pushd "${SAAS_OPERATOR_DIR}"
-# if branch doesn't exist, checkout a new branch based on staging
-if ! git show-ref --verify --quiet refs/heads/${BRANCH}; then
-    git checkout -b "${BRANCH}" --track origin/staging
+if [[ "${RELEASE_BRANCHED_BUILDS}" ]]; then
+  git clone ${GIT_PATH} "$SAAS_OPERATOR_DIR"
+  pushd "${SAAS_OPERATOR_DIR}"
+  # if branch doesn't exist, checkout a new branch based on staging
+  if ! git show-ref --verify --quiet refs/heads/${BRANCH}; then
```

Comment:
I do have a question on this.
This repo is freshly cloned on L91, which means there will be no local branch. If BRANCH=production, then this check will fail and checkout a new branch based on staging. But remote branch origin/production is actually exist, so the later git push process will fail. So it should check ref on remote/origin?